sweep/walletsweep.go

```diff
@@ -66,3 +70,211 @@ func DetermineFeePerKw(feeEstimator lnwallet.FeeEstimator,
+		// current UTXO state of the wallet.
+		utxos, err := utxoSource.ListUnspentWitness(
+			1, math.MaxInt32,
+		)
+		if err != nil {
+			return err
+		}
+
+		// We'll now lock each UTXO to ensure that other callers don't
+		// attempt to use these UTXOs in transactions while we're
+		// crafting out sweep all transaction.
+		for _, utxo := range utxos {
+			outpointLocker.LockOutpoint(utxo.OutPoint)
+		}
+
+		allOutputs = append(allOutputs, utxos...)
+
+		return nil
+	})
+	if err != nil {
+		// If we failed at all, we'll unlock any outputs selected just
+		// in case we had any lingering outputs.
+		unlockOutputs()
+
+		return nil, fmt.Errorf("unable to fetch+lock wallet "+
+			"utxos: %v", err)
+	}
+
+	// Now that we've locked all the potential outputs to sweep, we'll
+	// assemble an input for each of them, so we can hand it off to the
+	// sweeper to generate and sign a transaction for us.
+	var inputsToSweep []Input
+	for _, output := range allOutputs {
+		// We'll consult the utxoSource for information concerning this
+		// outpoint, we'll need to properly populate a signDescriptor
+		// for this output.
+		outputInfo, err := utxoSource.FetchInputInfo(&output.OutPoint)
+		if err != nil {
+			return nil, err
+		}
+
+		// As we'll be signing for outputs under control of the wallet,
+		// we only need to populate the output value and output script.
+		// The rest of the items will be populated internally within
+		// the sweeper via the witness generation function.
+		signDesc := &lnwallet.SignDescriptor{
+			Output:   outputInfo,
+			HashType: txscript.SigHashAll,
+		}
+
+		pkScript := outputInfo.PkScript
+
+		// Based on the output type, we'll map it to the proper witness
+		// type so we can generate the set of input scripts needed to
+		// sweep the output.
+		var witnessType lnwallet.WitnessType
+		switch {
+
+		// If this is a p2wkh output, then we'll assume it's a witness
+		// key hash witness type.
+		case txscript.IsPayToWitnessPubKeyHash(pkScript):
+			witnessType = lnwallet.WitnessKeyHash
+
+		// If this is a p2sh output, then as since it's under control
+		// of the wallet, we'll assume it's a nested p2sh output.
+		case txscript.IsPayToScriptHash(pkScript):
+			witnessType = lnwallet.NestedWitnessKeyHash
+
+		// All other output types we count as unknown and will fail to
+		// sweep.
+		default:
+			unlockOutputs()
+
+			return nil, fmt.Errorf("unable to sweep coins, "+
+				"unknown script: %x", pkScript[:])
+		}
+
+		// Now that we've constructed the items required, we'll make an
+		// input which can be passed to the sweeper for ultimate
+		// sweeping.
+		input := MakeBaseInput(&output.OutPoint, witnessType, signDesc, 0)
+		inputsToSweep = append(inputsToSweep, &input)
+	}
+
+	// Next, we'll map the fee preference to an actual absolute fee, and
+	// convert the delivery addr to a pkScript that we can use to create
+	// the sweep transaction.
+	feePerKw, err := DetermineFeePerKw(feeEstimator, FeePreference{
```

Comment:
Is this call needed? The fee rate passed into the method has already been through this.

Reply:
Indeed, as we already do this mapping in the RPC server, removed in the latest diff.

Reply:
Changes not pushed?

Reply:
Odd...fixed on my end. PTAL